Client/src/components/contanier/staff/Courses/ViewCourse-Details/index.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import "./index.css";
+import { Redirect } from 'react-router-dom';
+import Table from "../../../../common/Table";
+import Loading from '../../../../Layout/Loading'
+
+export default class ViewCourseDetails extends Component {
+  state = {
+    headings: null,
+    rows: null,
+    Error: null
+  }
+  componentDidMount() {
+    const { id } = this.props.match.params;
+    if (id) {
+      fetch(`/api/v1/course/${id}`)
+        .then(res => res.json())
+        .then(res => {
+          if (res.data) {
+            const rows = res.data;
+            let rowContent = [];
+            rows.map(row => {
+              rowContent.push([row.id, row.title, row.publish_date, row.target_level, row.number_of_student]);
+            })
+            this.state.headings = [
+              "Course Name",
+              "Date Of Publish",
+              "Target Level",
+              "Percentage Of Total Numbers"
+            ];
+            return this.setState({ rows: rowContent, Error: '' })
+          }
+          if (res.error) this.setState({ Error: res.error })
+        })
+        .catch(err => this.setState({ Error: 'Something happen error' }))
+    }
+    else return <Redirect to='/courses' />
```

Comment:
@Fatmasiam Like I said in my comment above, you'll always have an `id` so this `else` will never be reached, but just FYI anyways, `componentDidMount()` does not render anything, so `return <Redirect to='courses' />` wouldn't do anything, it would just be ignored.  You'd need to put a `<Redirect>` inside your render function for the redirect to happen.